sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeFormatter.scala

```diff
@@ -33,26 +33,37 @@ sealed trait DateTimeFormatter {
   def format(us: Long): String
 }
 
+trait FormatterUtils {
+  def zoneId: ZoneId
+  def buildFormatter(pattern: String, locale: Locale): java.time.format.DateTimeFormatter = {
+    new DateTimeFormatterBuilder()
+      .appendPattern(pattern)
+      .parseDefaulting(ChronoField.YEAR_OF_ERA, 1970)
+      .parseDefaulting(ChronoField.MONTH_OF_YEAR, 1)
+      .parseDefaulting(ChronoField.DAY_OF_MONTH, 1)
+      .parseDefaulting(ChronoField.HOUR_OF_DAY, 0)
+      .parseDefaulting(ChronoField.MINUTE_OF_HOUR, 0)
+      .parseDefaulting(ChronoField.SECOND_OF_MINUTE, 0)
+      .toFormatter(locale)
+  }
+  def toInstant(temporalAccessor: TemporalAccessor): java.time.Instant = {
```

Comment:
mark it as `protected` if it's only used in child

Reply:
maybe call it `toInstantWithDefaultTimeZone`?

Reply:
Strictly speaking it uses non-default time zone, what about `toInstantWithZoneId`?

Reply:
sure